### PR TITLE
make jl_init more friendly to embedded targets

### PR DESCRIPTION
**Pull request by vtjnash**
this throws the error messages directly when the problem occurs,
rather than waiting until late to throw a generic "System image file not found" message

### CODE REVIEW

File: base/replutil.jl [hunk 1]
```diff
@@ -151,6 +151,21 @@ function show(io::IO, ::MIME"text/plain", s::String)
     end
 end
 
+function show(io::IO, ::MIME"text/plain", opt::JLOptions)
```
Comment (tkelman) [inline base/replutil.jl]: do there need to be 2 copies of this code?

Comment (vtjnash) [inline base/replutil.jl]: because they aren't the same?

Comment (JeffBezanson) [inline base/replutil.jl]: Not identical, but they look extremely similar to me. I don't think we really need both.

**Comment by vtjnash** [comment]
CI failures were the known (spawn interrupt and windows libgit2) issues

File: base/replutil.jl (hunk 1 above)
Comment (tkelman) [inline base/replutil.jl]: was not addressed

File: test/misc.jl [hunk 2]
```diff
@@ -514,9 +514,19 @@ if is_windows()
     end
 end
 
-optstring = sprint(show, Base.JLOptions())
-@test startswith(optstring, "JLOptions(")
-@test endswith(optstring, ")")
+let optstring = stringmime(MIME("text/plain"), Base.JLOptions())
+    @test startswith(optstring, "JLOptions(\n")
+    @test !contains(optstring, "Ptr")
```
Comment (tkelman) [inline test/misc.jl]: seems to fail once in a while? https://travis-ci.org/JuliaLang/julia/jobs/251190331

**Pull request merged**